presentation/src/main/java/me/androidbox/presentation/task/viewmodel/TaskDetailViewModel.kt

```diff
@@ -32,9 +100,90 @@ class TaskDetailViewModel @Inject constructor(
                     )
                 }
             }
-            is TaskDetailScreenEvent.OnShowDeleteEventAlertDialog -> TODO()
-            TaskDetailScreenEvent.OnSaveTaskDetails -> TODO()
-            is TaskDetailScreenEvent.OnShowAlarmReminderDropdown -> TODO()
+            is TaskDetailScreenEvent.OnShowDeleteEventAlertDialog -> {
+                _taskDetailScreenState.update { taskDetailScreenState ->
+                    taskDetailScreenState.copy(
+                        shouldShowDeleteAlertDialog = taskDetailScreenEvent.shouldShowDeleteAlertDialog
+                    )
+                }
+            }
+            is TaskDetailScreenEvent.OnSaveTaskDetails -> {
+                insertTaskDetails(taskDetailScreenEvent.taskId)
+            }
+            is TaskDetailScreenEvent.OnShowAlarmReminderDropdown -> {
+                _taskDetailScreenState.update { taskDetailScreenState ->
+                    taskDetailScreenState.copy(
+                        shouldOpenDropdown = taskDetailScreenEvent.shouldOpen
+                    )
+                }
+            }
+        }
+    }
+
+    private fun insertTaskDetails(taskId: String) {
+        val startDateTime = AlarmReminderProvider.getCombinedFromDate(taskDetailScreenState.value.from)
+        val remindAt = AlarmReminderProvider.getRemindAt(taskDetailScreenState.value.alarmReminderItem, startDateTime)
+
+        val task = Task(
+            id = taskId,
+            title = taskDetailScreenState.value.taskTitle,
+            description = taskDetailScreenState.value.taskDescription,
+            startDateTime = taskDetailScreenState.value.from.toEpochSecond(),
+            remindAt = remindAt.toEpochSecond(),
+            isDone = taskDetailScreenState.value.isDone)
+
+        viewModelScope.launch {
+            taskRepositoryImp.insertTask(task)
+
+            val alarmItem = task.toAlarmItem()
+            alarmScheduler.scheduleAlarmReminder(alarmItem)
```

Comment:
Personally, I handled all the alarm stuff in the Repository. Whenever I deal with Services, AlarmManager, and "data" stuff from Android, I tend to access it through a Repository

Reply:
@MKiperszmid Updated